spec/requests/api/v1/targets/show_spec.rb

```diff
@@ -1,8 +1,9 @@
 RSpec.describe 'GET /api/v1/Targets/:id', type: :request do
-  subject { get api_v1_target_url(target_id), as: :json }
+  subject { get api_v1_target_url(target_id), headers: auth_headers, as: :json }
+  let!(:user) { create(:user) }
+  let(:topic) { create(:topic) }
 
   context 'when the request is valid' do
-    let(:topic) { create(:topic) }
     let(:target) { create(:target, topic: topic) }
     let(:target_id) { target.id }
```

Comment:
I think you can remove this variable and just call `target.id` whenever you need.